mlos_bench/mlos_bench/run.py

```diff
@@ -149,13 +149,14 @@ def _run(env_context: Environment, opt: Optimizer,
         opt.register(trial.tunables, Status.FAILED)
         return
 
+    (status, results) = env_context.run()  # Block and wait for the final result.
+    _LOG.info("Results: %s :: %s\n%s", trial.tunables, status, results)
+
     # In async mode (TODO), poll the environment for status and telemetry
     # and update the storage with the intermediate results.
-    (status, telemetry) = env_context.status()
+    (_, telemetry) = env_context.status()
```

Comment:
Why is status dropped here?
We update it in the next line, right?

Reply:
yes, we just use the status from `.run()` that we get a few lines above. Once we switch to the async mode, we'll query the `.status()` while `.run()` is still in progress, and *then* we'll use the status from it

Reply:
Maybe add a small comment to that effect.